server/src/test/java/org/elasticsearch/index/query/RangeQueryBuilderTests.java

```diff
@@ -601,4 +601,28 @@ public void testTypeField() throws IOException {
         builder.doToQuery(createShardContext());
         assertWarnings(QueryShardContext.TYPES_DEPRECATION_MESSAGE);
     }
+
+    /**
+     * Range queries should generally be cacheable, at least the ones we create randomly.
+     * This test makes sure we also test the non-cacheable cases regularly.
+     */
+    @Override
+    public void testCacheability() throws IOException {
+        RangeQueryBuilder queryBuilder = createTestQueryBuilder();
+        QueryShardContext context = createShardContext();
+        QueryBuilder rewriteQuery = rewriteQuery(queryBuilder, new QueryShardContext(context));
+        assertNotNull(rewriteQuery.toQuery(context));
+        assertTrue("query should be cacheable: " + queryBuilder.toString(), context.isCacheable());
+
+        // queries on date fields using "now" should not be cached
+        queryBuilder = new RangeQueryBuilder(randomFrom(DATE_FIELD_NAME, DATE_RANGE_FIELD_NAME, DATE_ALIAS_FIELD_NAME));
+        Instant now = Instant.now();
```

Comment:
We don't use the current time to create the `QueryShardContext` in the AbstractBuilderTestCase:
https://github.com/elastic/elasticsearch/blob/29992cff67d4e4d1c105daf66aa8f27329d669c0/test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java#L160
Since we disallow a `from` value that is after the `to` value I think you can use a range query that search for data before `now` (so only set the `to`) ?